kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -497,10 +497,16 @@ synchronized void registerComponentOCD(String metatypePid, Tocd ocd, boolean isF
     }
 
     synchronized void registerComponentConfiguration(String pid, String servicePid, String factoryPid) {
+        
```

Comment:
If a component has not a servicePid or a factoryPid, the method returns and the component is not added to m_allActivatedPids. The method that assigns the pid to the servicePid is updateWithDefaultConfiguration, but this is never executed because of the initial check on the pid/servicePid.
We found this bug when a component configuration, without kura.service.pid, was loaded from the snapshot_0.